Proxy/Pattern.ts

```diff
@@ -57,4 +57,57 @@ export namespace Pattern {
 		}
 		return result
 	}
+
+	export function getInPlain(data: any, pattern: Pattern, matchDelimiter: string, nameValueDelimiter: string): string {
+		let result: string | null
+		const regex = new RegExp(
+			(typeof pattern == "string" ? pattern : pattern.source.toString()) +
+				nameValueDelimiter +
+				"(.*?)" +
+				"(?=" +
+				matchDelimiter +
+				"|$)"
+		)
+
+		const matched = data.match(regex)
+		if (matched == null) {
+			throw Error
+		} else {
+			result = matched[0].toString()
+		}
+
+		return result != null ? result.substring(result.indexOf(nameValueDelimiter) + nameValueDelimiter.length) : ""
```

Comment:
is == null the correct comparison? 

and, result can't be null


also as a minor style thing - the else isnt needed

Reply:
maybe result could be null but toString would npe